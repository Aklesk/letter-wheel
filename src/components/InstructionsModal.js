import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import Modal from 'react-native-modalbox';

import InstructionsModalStyle from '../styles/InstructionsModalStyle';

const InstructionsModal = ({ isOpen, onClosed }) => (
  <Modal
    backButtonClose
    entry="bottom"
    isOpen={isOpen}
    onClosed={onClosed}
    position="center"
    style={InstructionsModalStyle.modal}
    swipeToClose
  >
    <View style={InstructionsModalStyle.container}>
      <Text style={InstructionsModalStyle.text}>
        {'Target Words is a word-finding game. Select letters from the grid to ' +
        'make words, then click "submit" to see if you found one! Each letter can be used once ' +
        'per word, and every word must contain the center letter.'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'Try variations on valid words you find: "Word", "wording", "words", ' +
        '"wordings", and "worded" are all valid, for example, and both American and British spellings ' +
        'are valid.'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'For a challenge, try timed mode. Time is added for each correct word, but the ' +
        'clock is relentless!'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <View>
        <Text style={InstructionsModalStyle.text}>
          {'Speed tips:'}
        </Text>
        <Text style={InstructionsModalStyle.text}>
          {'  • Long-press letters to submit'}
        </Text>
        <Text style={InstructionsModalStyle.text}>
          {'  • Long-press backspace to clear'}
        </Text>
      </View>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'Good luck!'}
      </Text>
    </View>
  </Modal>
);

InstructionsModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired,
};

export default InstructionsModal;
