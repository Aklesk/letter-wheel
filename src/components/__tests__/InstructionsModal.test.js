import React from 'react';
import { mount } from 'enzyme';

import InstructionsModal from '../InstructionsModal';

describe('Instructions Modal', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    const render = mount((
      <InstructionsModal
        isOpen={false}
        onClosed={jest.fn()}
      />
    ));
    expect(render).toMatchSnapshot();
    render.setProps({ isOpen: true });
    expect(render).toMatchSnapshot();
  });

  it('calls onClosed when closed once all animation has finished', () => {
    const render = mount((
      <InstructionsModal
        isOpen
        onClosed={jest.fn()}
      />
    ));
    render.setProps({ isOpen: false });
    expect(render.props().onClosed).toHaveBeenCalledTimes(0);
    jest.runAllTimers();
    expect(render.props().onClosed).toHaveBeenCalledTimes(1);
    render.setProps({ isOpen: true });
    expect(render.props().onClosed).toHaveBeenCalledTimes(1);
  });
});
