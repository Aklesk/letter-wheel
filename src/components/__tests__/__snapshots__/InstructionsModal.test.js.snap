// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions Modal matches the snapshot 1`] = `
<InstructionsModal
  isOpen={false}
  onClosed={[MockFunction]}
>
  <ModalBox
    animationDuration={400}
    backButtonClose={true}
    backdrop={true}
    backdropColor="black"
    backdropContent={null}
    backdropOpacity={0.5}
    backdropPressToClose={true}
    coverScreen={false}
    easing={[Function]}
    entry="bottom"
    isOpen={false}
    keyboardTopOffset={22}
    onClosed={[MockFunction]}
    position="center"
    startOpen={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "width": 263.52941176470586,
      }
    }
    swipeThreshold={50}
    swipeToClose={true}
  >
    <View>
      <react-native-mock />
    </View>
  </ModalBox>
</InstructionsModal>
`;

exports[`Instructions Modal matches the snapshot 2`] = `
<InstructionsModal
  isOpen={true}
  onClosed={[MockFunction]}
>
  <ModalBox
    animationDuration={400}
    backButtonClose={true}
    backdrop={true}
    backdropColor="black"
    backdropContent={null}
    backdropOpacity={0.5}
    backdropPressToClose={true}
    coverScreen={false}
    easing={[Function]}
    entry="bottom"
    isOpen={true}
    keyboardTopOffset={22}
    onClosed={[MockFunction]}
    position="center"
    startOpen={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "width": 263.52941176470586,
      }
    }
    swipeThreshold={50}
    swipeToClose={true}
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "zIndex": 2,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <react-native-mock>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <react-native-mock>
            <TouchableWithoutFeedback
              onPress={[Function]}
            >
              <react-native-mock>
                <AnimatedComponent
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "opacity": AnimatedValue {
                          "_animation": null,
                          "_children": Array [],
                          "_listeners": Object {},
                          "_offset": 0,
                          "_value": 0,
                        },
                      },
                    ]
                  }
                >
                  <View>
                    <react-native-mock>
                      <View
                        style={
                          Array [
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "backgroundColor": "black",
                              "opacity": 0.5,
                            },
                          ]
                        }
                      >
                        <react-native-mock />
                      </View>
                    </react-native-mock>
                  </View>
                </AnimatedComponent>
              </react-native-mock>
            </TouchableWithoutFeedback>
            <AnimatedComponent
              onLayout={[Function]}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                  },
                  Object {
                    "height": 768,
                    "width": 320,
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 263.52941176470586,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "translateY": AnimatedValue {
                          "_animation": null,
                          "_children": Array [],
                          "_listeners": Object {},
                          "_offset": 0,
                          "_value": 768,
                        },
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  },
                ]
              }
            >
              <View>
                <react-native-mock>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "borderRadius": 4.517647058823529,
                        "padding": 22.588235294117645,
                      }
                    }
                  >
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          Letter Wheel is a word-finding game. Select letters from the grid to make words, then click "submit" to see if you found one! Each letter can be used once per word, and every word must contain the center letter.
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          Try variations on valid words you find: "Word", "wording", "words", "wordings", and "worded" are all valid, for example, and both American and British spellings are valid.
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          For a challenge, try timed mode. Time is added for each correct word, but the clock is relentless!
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <View>
                        <react-native-mock>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 20.32941176470588,
                              }
                            }
                          >
                            <react-native-mock>
                              Speed tips:
                            </react-native-mock>
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 20.32941176470588,
                              }
                            }
                          >
                            <react-native-mock>
                                • Long-press letters to submit
                            </react-native-mock>
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#000",
                                "fontSize": 20.32941176470588,
                              }
                            }
                          >
                            <react-native-mock>
                                • Long-press backspace to clear
                            </react-native-mock>
                          </Text>
                        </react-native-mock>
                      </View>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          Good luck!
                        </react-native-mock>
                      </Text>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
            </AnimatedComponent>
          </react-native-mock>
        </View>
      </react-native-mock>
    </View>
  </ModalBox>
</InstructionsModal>
`;
